EFCore/MAUI/Readme.md

```diff
@@ -0,0 +1,873 @@
+       return Ok();
+   }
+   ```
+
+2. Add a method used to archive a post to the .NET MAUI application's data service.
+
+   _Services/IDataStore.cs_:
+
+   ```cs
+   public interface IDataStore<T> {
+       // ...
+       Task ArchivePostAsync(T post);
+   }
+   ```
+
+   _Services/WebAPIService.cs_:
+   
+   ```cs
+   public class WebAPIService : IDataStore<Post> {
+      // ...
+       public async Task ArchivePostAsync(Post post) {
+           var httpResponseMessage = await HttpClient.PostAsync($"{_apiUrl}CustomEndPoint/Archive", new StringContent(JsonSerializer.Serialize(post), Encoding.UTF8, ApplicationJson));
+           if (httpResponseMessage.IsSuccessStatusCode) {
+               await Shell.Current.DisplayAlert("Success", "This post is saved to disk", "Ok");
+           }
+           else {
+               await Shell.Current.DisplayAlert("Error", await httpResponseMessage.Content.ReadAsStringAsync(), "Ok");
+           }
+       }
+   }
+   ```
+
+3. Configure the detail view.
+
+   _ViewModels/ItemDetailViewModel.cs_:
+
+   ```cs
+    public class ItemDetailViewModel : BaseViewModel, IQueryAttributable {
+        public ItemDetailViewModel() => ArchiveCommand = new Command(OnArchive);
+
+        private async void OnArchive(object obj) {
+            if (!await DataStore.UserCanCreatePostAsync()) {
+                await Shell.Current.DisplayAlert("Error", "Permission denied", "OK");
+            }
+            else {
+                await DataStore.ArchivePostAsync(Post);
+            }
+        }
+
+        public Command ArchiveCommand { get; }
+    }
+   ```
+
+   _Views/ItemDetailPage.xaml_
+   ```xml
+   <ContentPage.ToolbarItems >
+       <ToolbarItem Text="Archive" Command="{Binding ArchiveCommand}"></ToolbarItem>
+   </ContentPage.ToolbarItems>
+   ```
+
+![](../../images/MAUI/Archive.png)
+
+### Add reporting functionality
+
+The [XAF Reports module](https://docs.devexpress.com/eXpressAppFramework/113591/shape-export-print-data/reports/reports-v2-module-overview?p=netframework) is a [Universal Subscription](https://www.devexpress.com/subscriptions/universal.xml) feature that you can use to easily integrate [DevExpress Reports](https://www.devexpress.com/subscriptions/reporting/) into your backend Web API service. Skip this step if you are using the Web API service as a part of the _DevExpress .NET App Security Library_ & Web API service free offer.
+
+Follow the steps bellow to create and initialize a report:
+
+1. Add the Report component using the Visual Studio New Item wizard.
+2. Drag and drop a CollectionDataSource component from the Visual Studio toolbox and change its ObjectTypeName to `WebAPI.BusinessObjects.Post`.
+3. Drag & drop all discovered fields from the Field List window onto the Report details surface.
+
+   ![](../../images/MAUI/ReportDesigner.png) 
+4. Use a predefined reports updater to initialize the report (Module.cs).
```

Comment:
Why don't you indicate the file in the same way that you did in other examples: separate line before the code snippet?

Reply:
```suggestion
4. Use a predefined reports updater to initialize the report.

_Module.cs_:
```